src/transport/manager/mod.rs

```diff
@@ -267,13 +267,13 @@ impl TransportManager {
         let peers = Arc::new(RwLock::new(HashMap::new()));
         let (cmd_tx, cmd_rx) = channel(256);
         let (event_tx, event_rx) = channel(256);
-        let listen_addresses = Arc::new(RwLock::new(HashSet::new()));
+        let listen_addresses = PublicAddresses::new(local_peer_id);
         let handle = TransportManagerHandle::new(
             local_peer_id,
             peers.clone(),
             cmd_tx,
             supported_transports,
-            Arc::clone(&listen_addresses),
+            listen_addresses.clone(),
```

Comment:
Is `clone()` needed here?

Reply:
Yep, we need a reference in the transport manager as well